rust/datafusion/src/datasource/parquet.rs

```diff
@@ -262,20 +263,55 @@ impl ParquetFile {
                                 is_nullable,
                             )?
                         }
-                        ColumnReader::Int32ColumnReader(ref mut r) => {
-                            ArrowReader::<Int32Type>::read(
+                        ColumnReader::Int32ColumnReader(ref mut r) => match dt {
+                            DataType::Date32(DateUnit::Millisecond) => {
```

Comment:
If this condition's ever true, then it might be that we've incorrectly implemented the spec in the parquet crate. @sunchao am I misinterpreting the logical types spec?

Reply:
@nevi-me what do you mean? int32 when used as date, represents the number of days since epoch. See https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#datetime-types.

Reply:
Sorry, I meant that the match arm for date 32 with milli should never evaluate to true

Reply:
Yes, agreed. This should never happen.